engineering_tools/carma_script_extensions/cloud-sim

```diff
@@ -0,0 +1,348 @@
+
+    local PUBLIC_DNS=$(aws ec2 describe-instances --instance-ids ${DEV_MACHINE_ID} --output text --query Reservations[].Instances[].PublicDnsName)
+
+    # TODO determine cloud 9 address
+    local CLOUD9_ADDRESS_BASE="https://us-east-1.console.aws.amazon.com/cloud9/ide"
+    # 1. Extract cloud9 name from stack details
+    local CLOUD9_ENVS=$(aws cloud9 list-environments --output text | sed "s|ENVIRONMENTIDS\s||")
+    # 2. Determine cloud9 environment id
+    local CLOUD9_NAME="RobotWorkshop-${STACK_NAME}"
+    local CLOUD9_ID=$(aws cloud9 describe-environments --output text --environment-ids $(echo $CLOUD9_ENVS) --query "environments[?name=='${CLOUD9_NAME}'].id")
+    # 3. Update URL
+    local CLOUD9_ADDRESS="${CLOUD9_ADDRESS_BASE}/${CLOUD9_ID}"
+
+    # Save the stack name for cleanup
+    echo "STACK_NAME=${STACK_NAME}" > ${CARMA_CLOUD_SIM_STACK_PATH}
+    
+    # Print for user
+    echo "##############################"
+    echo ""
+    echo "AWS Stack ${STACK_NAME} Launch Suceeded"
+    echo ""
+    echo "EC2 Instance Address: ${PUBLIC_DNS}"
+    echo ""
+    echo "Cloud9 IDE Address: ${CLOUD9_ADDRESS}"
+    echo ""
+    echo "##############################"
+}
+
+cloud-sim__stop() {
+
+    if [ ! -f "$CARMA_CLOUD_SIM_STACK_PATH" ]; then
+        echo "No stack cache file exists cannot determine name. Either the stack was never started or an error has occured and you must manually delete the stack on AWS."
+        exit 1
+    fi
+
+    local STACK_NAME=$(cat ${CARMA_CLOUD_SIM_STACK_PATH} | grep "STACK_NAME=" | sed "s|STACK_NAME=||")
+
+    if [ -z "$STACK_NAME" ]; then
+        echo "STACK_NAME not found in cache file. Either the stack was never started or an error has occured and you must manually delete the stack on AWS."
+        exist -1
+    fi
+
+    aws --region us-east-1 cloudformation delete-stack --stack-name "${STACK_NAME}"
+}
+
+cloud-sim__status() {
+    if [ ! -f "${CARMA_CLOUD_SIM_STACK_PATH}" ]; then
+        echo "No stack cache file exists cannot report status. If you expected a running stack this might indicate an error."
+        exit 1
+    fi
+
+    local STACK_NAME=$(cat ${CARMA_CLOUD_SIM_STACK_PATH} | grep "STACK_NAME=" | sed "s|STACK_NAME=||")
+
+    if [ -z "${STACK_NAME}" ]; then
+        echo "STACK_NAME not found in cache file. If you expected a running stack this might indicate an error."
+        exit 1
+    fi
+
+    aws cloudformation describe-stack-resources --stack-name "${STACK_NAME}"
+}
+
+
+cloud-sim__help() {
+    cat <<HELP
+-------------------------------------------------------------------------------
+| USDOT FHWA STOL CARMA                                              |
+-------------------------------------------------------------------------------
+
+Please enter one of the following commands when using the cloud-sim extension:
+    carma
+        cloud-sim: 
+            config
```

Comment:
Should it be specificed (or provided an example) of how to use this? Can multiple arguments be specified at the same time?

Reply:
They can. I'll clarify that in the description.